File: src/main/java/me/pcasaes/hexoids/core/domain/index/PlayerSpatialIndex.java
package me.pcasaes.hexoids.core.domain.index;

import me.pcasaes.hexoids.core.domain.model.Player;

/**
 * Allows us to search for player using a spatial index.
 * The domain does not provide an implementation.
 * A simple implementation would return all players in {@link me.pcasaes.hexoids.core.domain.model.Players}
 */
public interface PlayerSpatialIndex {
    Iterable<Player> search(float x1, float y1, float x2, float y2, float distance);
}

File: src/main/kotlin/me/pcasaes/hexoids/core/domain/index/PlayerSpatialIndex.kt
package me.pcasaes.hexoids.core.domain.index

import me.pcasaes.hexoids.core.domain.model.Player

/**
 * Allows us to search for player using a spatial index.
 * The domain does not provide an implementation.
 * A simple implementation would return all players in [me.pcasaes.hexoids.core.domain.model.Players]
 */
fun interface PlayerSpatialIndex {
    fun search(x1: Float, y1: Float, x2: Float, y2: Float, distance: Float): Iterable<Player>
}
